src/cr/cube/collator.py

```diff
@@ -459,7 +459,12 @@ def _subtotal_idxs(self):
         keys, nans = [], []
         for i, val in enumerate(subtotal_values):
             neg_idx = i - n_values
-            group = nans if np.isnan(val) else keys
+            # --- TODO: this try except does not seem idiomatic, but np.isnan fails
```

Comment:
I don't like this code, but wasn't sure the best way to accomplish it.

Reply:
Hm, now that I think about it, I don't see why only the subtotals behave this way, body values can also be nan. For example, the scale mean marginal would be nan if a row only has counts in columns that do not have a numeric value assigned.